aspnetcore/fundamentals/app-state.md

```diff
@@ -309,6 +309,10 @@ Use [Dependency Injection](xref:fundamentals/dependency-injection) to make data
   For example, a user stores a shopping cart in session. The user adds an item to the cart but the commit fails. The app doesn't know about the failure so it reports to the user that the item was added to their cart, which isn't true.
 
   The recommended approach to check for errors is to call `await feature.Session.CommitAsync();` from app code when the app is done writing to the session. `CommitAsync` throws an exception if the backing store is unavailable. If `CommitAsync` fails, the app can process the exception. `LoadAsync` throws under the same conditions where the data store is unavailable.
+  
+## SignalR and state
+
+SignalR apps should not use state to store information. SignalR apps can store per user state in `Context.Items` on the in the hub.
```

Comment:
```suggestion
SignalR apps should not use state to store information. SignalR apps can store per connection state in `Context.Items` in the hub.
```

Reply:
"SignalR apps should not use state to store information."
This statement sounds wrong and contradictory to the next statement.

Are you trying to say that you can't store non-static or non-singleton state inside your hub?

Reply:
@BrennanConroy  see https://github.com/aspnet/SignalR/issues/2139